subprojects/plugins/src/main/java/org/gradle/api/plugins/GroovyBasePlugin.java

```diff
@@ -120,7 +122,9 @@ private void configureSourceSetDefaults() {
                 JvmPluginsHelper.configureForSourceSet(sourceSet, groovySource, compile, compile.getOptions(), project);
                 compile.setDescription("Compiles the " + sourceSet.getName() + " Groovy source.");
                 compile.setSource(groovySource);
-                compile.getJavaLauncher().convention(getToolchainTool(project, JavaToolchainService::launcherFor));
+                Provider<JavaLauncher> javaLauncherConvention = getToolchainTool(project, JavaToolchainService::launcherFor)
```

Comment:
This is something which we need to follow up on, right?

Reply:
That's correct